### PR TITLE
Fix cart refresh after voucher removal

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | See below
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | maybe, see below
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/12969
| How to test?  | See fixed ticket

## Description

**This PR does 2 things:**
1) I removed some duplicated code between `cart-detailed-totals.tpl` and `cart-summary-totals.tpl`
2) I extracted the subtotals section from `cart-summary.tpl` into a dedicated block wrapped in a new div with class `cart-summary-subtotals-container` so that I could use the ajax reloading feature for cart with this section: basically when the cart is updated, the whole section within this `<div>` is re-rendered and replaced using javascript. Not pretty, but it works.

**Some things I wanted to do but could not 😢 :**
- use a single block for `cart-summary.tpl` and `cart-detailed-totals.tpl` because [code is almost identical](https://github.com/PrestaShop/PrestaShop/blob/develop/themes/classic/templates/checkout/_partials/cart-detailed-totals.tpl#L28) minor some `{if}` however if I introduce a `{if}` then it's hard to compute this block as stand-alone for ajax reloading 😭 so I did not do it
- use `cart-summary-subtotals` as a class name for the wrapping div, following other similar block naming conventions, because this class name is already used within the content of this div, so I used `cart-summary-subtotals-container` as a fallback

## Possible BC break
I added the div with `cart-summary-subtotals-container` in order to be able to use ajax reloading on it. This changes the DOM structure of the page where `cart-summary.tpl` is used. Kind of a BC but should be safe if people do not target it with bad XSS selectors.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13370)

### CODE REVIEW

File: themes/classic/templates/checkout/_partials/cart-detailed-totals.tpl [hunk 1]
```diff
@@ -47,29 +47,9 @@
     {/foreach}
   </div>
 
-  <div class="card-block cart-summary-totals">
```
Comment (Quetzacoalt91) [inline themes/classic/templates/checkout/_partials/cart-detailed-totals.tpl:50]: Can we keep this class? It may be used by themes using classic as parent.

Comment (Quetzacoalt91) [reply themes/classic/templates/checkout/_partials/cart-detailed-totals.tpl:50]: we could expect something like

```tpl
   {block name='cart_summary_totals'} 
     <div class="card-block cart-summary-totals"> 
       {include file='checkout/_partials/cart-summary-totals.tpl' cart=$cart} 
     </div> 
   {/block} 
```

File: themes/classic/templates/checkout/_partials/cart-summary-subtotals.tpl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *}
+
+<div class="card-block cart-summary-subtotals-container">
```
Comment (Quetzacoalt91) [inline themes/classic/templates/checkout/_partials/cart-summary-subtotals.tpl:26]: Following  my previous comment, we would get here
```suggestion
<div class="cart-summary-subtotals-container">
```

**Review by Quetzacoalt91 — COMMENTED** [review]
One comment regarding a possible BC Break

File: themes/classic/templates/checkout/_partials/cart-detailed-totals.tpl (hunk 1 above)
Comment (matks) [reply themes/classic/templates/checkout/_partials/cart-detailed-totals.tpl:50]: This class is already used in `checkout/_partials/cart-summary-totals.tpl`

https://github.com/PrestaShop/PrestaShop/blob/develop/themes/classic/templates/checkout/_partials/cart-summary-totals.tpl#L25

**Comment by matks** [comment]
I ... dont remember why I closed this PR 😅

**Comment by jolelievre** [comment]
> I ... dont remember why I closed this PR 😅

rage quit? 😄

File: themes/classic/templates/checkout/_partials/cart-detailed-totals.tpl (hunk 1 above)
Comment (jolelievre) [reply themes/classic/templates/checkout/_partials/cart-detailed-totals.tpl:50]: yes except you changed `card-block cart-summary-totals` with `card-block cart-summary-totals-container` so it actually changes your block class and potentialy css
I'm aware the class is also used in the loop..

but maybe you could use `.card-block.cart-summary-totals` as a selector or add an id `id=cart-summary-totals-container` or `id=cart-summary-totals-list` instead (unless the block is displayed several times in the page and you need to update all of them?)

**Comment by PierreRambaud** [comment]
@jolelievre Thinking about someone? :trollface:

**Comment by jolelievre** [comment]
> @jolelievre Thinking about someone? :trollface:

Nooo 😅

**Review by jolelievre — APPROVED** [review]
Hello @matks I tried modifying your PR, but I understood why you had to use the container
My bad! Thanks for the fix I simply rebased and rebuilt assets Ready for QA

**Comment by matks** [comment]
@Quetzacoalt91 Before merging, is it OK for you ? As we've seen together, there is a small BC break in HTML markup in this PR however I'm afraid it's necessary as @jolelievre confirmed it in order to fix the bug

**Comment by Quetzacoalt91** [comment]
Well, regarding your answer it seems I was wrong about the compatibility break (= no class deleted). This is fine for me.

**Review by Quetzacoalt91 — APPROVED** [review]

**Pull request merged**